adoptopenjdk-api-v3-persistance/src/main/kotlin/net/adoptopenjdk/api/v3/DownloadStatsInterface.kt

```diff
@@ -128,17 +179,25 @@ class DownloadStatsInterface(
                                         repoForDay.value.maxBy { it.date }!!
                                     }
                         }
-                        .filter { dockerRepo == null || it.repo == dockerRepo }
+                        .filter { (dockerRepo == null || it.repo == dockerRepo) && (featureVersion == null || it.feature_version == featureVersion) && (jvmImpl == null || it.jvm_impl == jvmImpl) }
```

Comment:
fairly long filter, should extract to method

Reply:
Not really a way to decrease the complexity here so I've just broken it down into separate lines